DocumentDataAPI/DocumentDataAPI/Controllers/DocumentController.cs

```diff
@@ -0,0 +1,208 @@
+        [HttpGet]
+        [Route("GetTotalDocumentCount")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+        public ActionResult<int> GetTotalDocumentCount()
+        {
+            try
+            {
+                int result = _repository.GetTotalDocumentCount();
+                return Ok(result);
+            }
+            catch (Exception e)
+            {
+                return Problem(e.Message);
+            }
+        }
+
+        /// <summary>
+        /// Retrieves a list of documents based on the source id.
+        /// </summary>
+        /// <response code="200">Success: A list of documents for the given source id.</response>
+        /// <response code="404">Not Found: Nothing is returned.</response>
+        /// <response code="500">Internal Server Error: a <see cref="ProblemDetails"/> describing the error.</response>
+        [HttpGet]
+        [Route("GetBySourceId/{id:int?}")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+        public ActionResult<IEnumerable<DocumentModel>> GetBySourceId(int id)
+        {
+            try
+            {
+                List<DocumentModel> result = _repository.GetBySource(id).ToList();
+                return result.Count == 0 ?
+                    NotFound() :
+                    Ok(result);
+            }
+            catch (Exception e)
+            {
+                _logger.LogError($"Unable to fetch document by source id: {id}.\n{e.Message}");
+                return Problem(e.Message);
+            }
+        }
+
+        /// <summary>
+        /// Retrieves a list of documents based on the author.
+        /// </summary>
+        /// <response code="200">Success: A list of documents by the given author.</response>
+        /// <response code="404">Not Found: Nothing is returned.</response>
+        /// <response code="500">Internal Server Error: a <see cref="ProblemDetails"/> describing the error.</response>
+        [HttpGet]
+        [Route("GetByAuthor")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+        public ActionResult<IEnumerable<DocumentModel>> GetByAuthor(string author)
+        {
+            try
+            {
+                List<DocumentModel> result = _repository.GetByAuthor(author).ToList();
+                return result.Count == 0 ?
+                    NotFound() :
+                    Ok(result);
+            }
+            catch (Exception e)
+            {
+                _logger.LogError($"Unable to fetch documents by author: {author}.\n{e.Message}");
```

Comment:
Samme som andre kommentarer
```suggestion
                _logger.LogError(e, "Unable to fetch documents by author: {author}.", author);
```